ppci/lang/c/semantics.py

```diff
@@ -25,6 +25,37 @@
 from .printer import expr_to_str, type_to_str
 
 
+class CSwitch:
+    """ Switch instruction context """
+    def __init__(self, typ):
+        self.typ = typ
+        self.default_seen = False
+        self.values = []
+        self.ranges = []
+
+    def add_value(self, val):
+        """ Add a case value, returns True if OK, False otherwise """
+        for low, high in self.ranges:
+            if val >= low and val <= high:
```

Comment:
Did you know that python has chained comparison logic?
In this case, you could use:
```python
if low <= val <= high:
```

Reply:
Yes, it's a neat feature. But I am careful with it because it pushes me to use it in other languages (eg C) where it is syntactically accepted but the result is not what is expected!